### PR TITLE
feat: support for custom certificate configuration

### LINKED ISSUE #516: Support for proxy and custom certificate configuration
**Motivation**

When using the TinyMist Typst VSCode extension in a proxy environment, I encountered an issue where document preview fails with the following error:

> typst_ts_compiler
> error: failed to download package: error trying to connect: invalid peer certificate: UnknownIssuer

This issue hinders the use of the extension in environments with strict network security, such as corporate networks.

**Description**

The issue appears to be related to certificate verification in a proxy environment. A similar problem has been reported in the Typst repository (see [Typst Issue #1678](https://github.com/typst/typst/issues/1678)), where the solution involved allowing environment variables or CLI arguments to bypass the certificate verification.

It would be beneficial if the TinyMist Typst extension could also support similar options, enabling users to configure the extension for proxy environments where certificate verification might fail.

**Examples/Questions**

- Is it possible to implement the support for environment variables or CLI arguments in TinyMist Typst to handle certificate verification issues in proxy environments?
- Would this feature allow users to set specific proxy-related configurations that could resolve this error?

Finally, this is my first time posting an issue, so I apologize if there are any mistakes or if I've overlooked anything.

Thank you for your time and for developing such a helpful tool!

### PR DESCRIPTION
**Pull request by ricOC3**
This pull request resolves #516. 

Thank you for the helpful advice in the issue discussion. 

Following suggestions, I have added support for specifying CA certificates of https request through the `--cert` option and the environment variable `TYPST_CERT` , as seen in Typst.

I’ve confirmed that the certification error is resolved by either passing the option as follows or setting `TYPST_CERT`:

```json
{
  "tinymist.typstExtraArgs": ["--cert=path/to/certificates"]
}
```

As this is my first time working with Rust, I apologize if there are any mistakes in my changes.

 I really appreciate your review and guidance.

### CODE REVIEW

File: crates/tinymist-world/src/lib.rs [hunk 1]
```diff
@@ -200,3 +230,221 @@ pub fn parse_source_date_epoch(raw: &str) -> Result<DateTime<Utc>, String> {
...
+            }, self.cert_path.as_deref())
+            .unwrap_or_default()
+        })
+    }
+}
+
+fn threaded_http<T: Send + Sync>(
+    url: &str,
+    f: impl FnOnce(Result<Response, reqwest::Error>) -> T + Send + Sync,
+    cert_path: Option<&Path>
```
Comment (Myriad-Dreamin) [inline crates/tinymist-world/src/lib.rs:428]: `f:` the callback argument should be the last argument, i.e. `cert_path:` should be placed before `f:`.

**Review by Myriad-Dreamin — COMMENTED** [review]
Wonderful! It looks generally great for me and I have left some comments.

> As this is my first time working with Rust, I apologize if there are any mistakes in my changes.

I wish you gain and learn more than having this feature and implementing this feature, :). learning valuable things like rust and communication are also important as well. I'm and 'll be delighted to review PRs like this.

File: crates/tinymist-world/src/lib.rs [hunk 2]
```diff
@@ -78,6 +90,16 @@ pub struct CompileOnceArgs {
         hide(true),
     )]
     pub creation_timestamp: Option<DateTime<Utc>>,
+
+    /// Path to the certification file
```
[the remark this answers is not in the document]
Comment (ricOC3) [reply crates/tinymist-world/src/lib.rs]: Thank you for the feedback. I’ve updated the description to that suggested.

File: crates/tinymist-world/src/lib.rs [hunk 3]
```diff
@@ -78,6 +90,16 @@ pub struct CompileOnceArgs {
...
+    ///
+    /// This value can be provided via the `--cert` option or the `TYPST_CERT` environment variable.
```
[the remark this answers is not in the document]
Comment (ricOC3) [reply crates/tinymist-world/src/lib.rs]: I’ve removed it accordingly.

File: crates/tinymist-world/src/lib.rs [hunk 4]
```diff
@@ -154,12 +177,19 @@ impl LspUniverseBuilder {
         entry: EntryState,
         font_resolver: Arc<FontResolverImpl>,
         inputs: ImmutDict,
+        cert_path: Option<PathBuf>,
     ) -> ZResult<LspUniverse> {
+
+        let mut registry = HttpsRegistry::default();
```
[the remark this answers is not in the document]
Comment (ricOC3) [reply crates/tinymist-world/src/lib.rs]: I’ve implemented `HttpsRegistry::new(cert_path)` and simplified `LspUniverseBuilder::build` .

File: crates/tinymist-world/src/lib.rs [hunk 5]
```diff
@@ -200,3 +230,221 @@ pub fn parse_source_date_epoch(raw: &str) -> Result<DateTime<Utc>, String> {
...
+    type Registry = HttpsRegistry;
+}
+
+/// The compiler universe in system environment.
+pub type TypstSystemUniverseExtend = CompilerUniverse<SystemCompilerFeatExtend>;
+/// The compiler world in system environment.
+pub type TypstSystemWorldExtend = CompilerWorld<SystemCompilerFeatExtend>;
+
+/// The http registry without typst.ts to implement more for tinymist.
+pub struct HttpsRegistry {
```
[the remark this answers is not in the document]
Comment (ricOC3) [reply crates/tinymist-world/src/lib.rs]: I’ve split the code into a separate file named `https.rs` as recommended. The relevant sections have been moved there.

File: crates/tinymist-world/src/lib.rs (hunk 1 above)
Comment (ricOC3) [reply crates/tinymist-world/src/lib.rs]: I’ve adjusted the argument order so that `f` is now the last argument.

**Comment by Myriad-Dreamin** [comment]
Thank you!

**Pull request merged**